### PR TITLE
Add fatal error on missing dependency

### PR DESCRIPTION
**Pull request by rubenwardy**
Minetest silently not loading the mod caused a security vulnerability on my server, because a mod wasn't loaded.

image

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -334,7 +334,22 @@ void Server::init()
 	std::vector<ModSpec> unsatisfied_mods = m_modmgr->getUnsatisfiedMods();
 	// complain about mods with unsatisfied dependencies
 	if (!m_modmgr->isConsistent()) {
-		m_modmgr->printUnsatisfiedModsError();
```
Comment (nerzhul) [inline src/server.cpp]: translations please for clients in single player

**Comment by rubenwardy** [comment]
You've both picked up on why this is WIP

**Comment by Fixer-007** [comment]
But how you disable those missing mods, they are not showing up in world configure...

**Comment by rubenwardy** [comment]
This isn't about missing mods, this is about mods with missing dependencies

**Comment by rubenwardy** [comment]
@paramat this is not related to CDB at all, and probably shouldn't be a blocker

**Comment by rubenwardy** [comment]
This is problematic, as it breaks mods which relay on this behaviour. For example, 3darmor has separate mods for sfinv and unified inventory support, and those mods will commonly be enabled all at the same time using "Enable Modpack"

**Comment by Wuzzy2** [comment]
WTF? Are you serious?

I absolutely agree with that Minetest should refuse to launch if there are missing dependencies.
Current behaviour is stupid and you yourself admitted it is a security vulnerability.

Screw mods that depend on broken behaviour. Technically, what you said here is not actually a problem of a mod, but a *modpack*.

I agree the dependencies of 3d_armor modpack are annoying but a security vulnerability should have higher priority.

The 3d_armor modpack problem seems to be just an usability problem to me. It could be fixed at the main menu level. One possible idea: If you click “Enable Modpack”, only enable mods for which the dependencies could be resolved with currently enabled mods. It's kind of stupid for the button to enable conflicting mods in the first place.

**Comment by paramat** [comment]
Maybe an issue should be opened about this so we don't forget?

**Pull request closed without merging**